tests/task_scheduler_test.c

```diff
@@ -198,7 +198,61 @@ static int test_scheduler_rejects_xthread_access(struct aws_allocator *alloc, vo
     return 0;
 }
 
+struct task_scheduler_reentrancy_args {
+    struct aws_task_scheduler *scheduler;
+    int executed;
+    struct task_scheduler_reentrancy_args *reentrant_args;
+};
+
+static void reentrancy_fn(void *arg) {
+
+    struct task_scheduler_reentrancy_args *reentrancy_args = (struct task_scheduler_reentrancy_args *)arg;
+
+    if (reentrancy_args->reentrant_args) {
+        struct aws_task task;
+        task.fn = reentrancy_fn;
+        task.arg = reentrancy_args->reentrant_args;
```

Comment:
The naming convention is a bit counterintuitive to me. struct task_scheduler_reentrancy_args has an element called reentrant_args. What does this mean, and can these names be made less cumbersome?

Reply:
it's a unit test, I don't really care that much, 

but it's self-referential so that's why it's named the same.